courseware/api.py

```diff
@@ -632,12 +632,20 @@ def enroll_in_edx_course_runs(user, course_runs):
         EdxApiEnrollErrorException: Raised if the underlying edX API HTTP request fails
         UnknownEdxApiEnrollException: Raised if an unknown error was encountered during the edX API request
     """
-    edx_client = get_edx_api_client(user)
+    username = None
+    if force_enrollment:
+        edx_client = get_edx_api_service_client()
+        username = user.username
+    else:
+        edx_client = get_edx_api_client(user)
     results = []
     for course_run in course_runs:
         try:
             result = edx_client.enrollments.create_student_enrollment(
-                course_run.courseware_id, mode=EDX_ENROLLMENT_PRO_MODE
+                course_run.courseware_id,
+                mode=EDX_ENROLLMENT_PRO_MODE,
+                username=username,
+                force_enrollment=force_enrollment,
```

Comment:
Just a question here, Can `force_enrollment` in edX collide with enrollment mode? I'm asking this because previously when we see an error for enrollment pro mode, we'd try enrolling in Audit. 

So, the question might be, if we send `force_enrollment` along it API call would it still throw an error on enrollment mode or this will even override enrollment mode thing as well?

Reply:
The `force_enrollment` parameter in edX does not affect the enrollment mode. Its purpose is to override the enrollment process and allow a user to be enrolled in a course, even if the enrollment date has passed.

If there is an issue with the Pro enrollment mode, it will still throw an error regardless of the use of the `force_enrollment` parameter.